Add REDIS_OPT_PREFER_UNSPEC (#1101)
redis/hiredis: README.md

| --- | --- |
| REDIS\_OPT\_NONBLOCK | Tells hiredis to make a non-blocking connection. |
| REDIS\_OPT\_REUSEADDR | Tells hiredis to set the [SO_REUSEADDR](https://man7.org/linux/man-pages/man7/socket.7.html) socket option |

> | REDIS\_OPT\_PREFER\_IPV4<br>REDIS\_OPT\_PREFER_IPV6<br>REDIS\_OPT\_PREFER\_UNSPEC | Informs hiredis to either prefer `IPV4` or `IPV6` when invoking [getaddrinfo](https://man7.org/linux/man-pages/man3/gai_strerror.3.html).  Note that both of the options may be set at once, which will cause hiredis to spcify `AF_UNSPEC` in the getaddrinfo call, which means both `IPV4` and `IPV6` addresses will be searched simultaneously.<br>For convenience `REDIS_OPT_PREFER_UNSPEC` can be used which has the same effect as setting both `REDIS_OPT_PREFER_IPV4` and `REDIS_OPT_PREFER_IPV6` flags. | 

Reviewer: I think we should promote using the UNSPEC option for those who want AF_UNSPEC behaviour. The note about setting both flags can be an implementation detail. No need to mention it IMHO.

Perhaps also mention what the default is.

Below suggestion also inserts "IP" in the unspec option name. (I think it's nice to include "IP" since it means unspecified IP version, not unspecified behaviour in general. :grin:)

```suggestion
| REDIS\_OPT\_PREFER\_IPV4<br>REDIS\_OPT\_PREFER_IPV6<br>REDIS\_OPT\_PREFER\_IP\_UNSPEC | Informs hiredis to either prefer IPv4 or IPv6 when invoking [getaddrinfo](https://man7.org/linux/man-pages/man3/gai_strerror.3.html).  `REDIS_OPT_PREFER_IP_UNSPEC` will cause hiredis to specify `AF_UNSPEC` in the getaddrinfo call, which means both IPv4 and IPv6 addresses will be searched simultaneously.<br>Hiredis prefers IPv4 by default. |
```
